nifi-nar-bundles/nifi-kudu-bundle/nifi-kudu-client-services/src/main/java/org/apache/nifi/kudu/KuduControllerService.java

```diff
@@ -0,0 +1,344 @@
+                addPredicate(scannerBuilder, kuduTable, parts[0], parts[1]);
+            }
+        }
+
+        if(!projectedColumnNames.isEmpty()){
+            scannerBuilder.setProjectedColumnNames(projectedColumnNames);
+        }
+
+        AsyncKuduScanner scanner = scannerBuilder.build();
+        while (scanner.hasMoreRows()) {
+            handler.handle(scanner.nextRows().join());
+        }
+    }
+
+    @VisibleForTesting
+    public void buildPartialRow(Schema schema, PartialRow row, Record record, List<String> fieldNames) {
+        for (String colName : fieldNames) {
+            int colIdx = this.getColumnIndex(schema, colName);
+            if (colIdx != -1) {
+                ColumnSchema colSchema = schema.getColumnByIndex(colIdx);
+                Type colType = colSchema.getType();
+
+                if (record.getValue(colName) == null) {
+                    row.setNull(colName);
+                    continue;
+                }
+
+                switch (colType.getDataType(colSchema.getTypeAttributes())) {
+                    case BOOL:
+                        row.addBoolean(colIdx, record.getAsBoolean(colName));
+                        break;
+                    case FLOAT:
+                        row.addFloat(colIdx, record.getAsFloat(colName));
+                        break;
+                    case DOUBLE:
+                        row.addDouble(colIdx, record.getAsDouble(colName));
+                        break;
+                    case BINARY:
+                        row.addBinary(colIdx, record.getAsString(colName).getBytes());
+                        break;
+                    case INT8:
+                        row.addByte(colIdx, record.getAsInt(colName).byteValue());
+                        break;
+                    case INT16:
+                        row.addShort(colIdx, record.getAsInt(colName).shortValue());
+                        break;
+                    case INT32:
+                        row.addInt(colIdx, record.getAsInt(colName));
+                        break;
+                    case INT64:
+                    case UNIXTIME_MICROS:
+                        row.addLong(colIdx, record.getAsLong(colName));
+                        break;
+                    case STRING:
+                        row.addString(colIdx, record.getAsString(colName));
+                        break;
+                    case DECIMAL32:
+                    case DECIMAL64:
+                    case DECIMAL128:
+                        row.addDecimal(colIdx, new BigDecimal(record.getAsString(colName)));
+                        break;
+                    default:
+                        throw new IllegalStateException(String.format("unknown column type %s", colType));
+                }
+            }
+        }
+    }
+
+    protected void addPredicate(AsyncKuduScanner.AsyncKuduScannerBuilder scannerBuilder, KuduTable kuduTable, String column, String value) {
```

Comment:
Is there a reason to only support `Equals` predicates?

Reply:
For now, I kept it only Equals predicates, I will create separate JIRA for adding more predicted support, right now I'm not sure about NIFI expressions supported for other predicates.